remote/setup_workspace.py

```diff
@@ -37,19 +39,45 @@ def replace_config(path: pathlib.Path, key: str, value: str) -> None:
     path.write_text(file_content)
 
 
+def replace_config_repositories(key: str, value: str) -> None:
+    replace_config(
+        REPOSITORIES_BASE_PATH
+        / "fr.obeo.dsl.viewpoint.collab"
+        / "repository.properties",
+        key,
+        value,
+    )
+
+
 def setup_repositories() -> None:
     t4c_repositories = os.getenv("T4C_REPOSITORIES", "").split(",")
     t4c_host = os.getenv("T4C_SERVER_HOST", "localhost")
     t4c_port = os.getenv("T4C_SERVER_PORT", 2036)
+    t4c_json = os.getenv("T4C_JSON", None)
 
-    for repo in t4c_repositories:
-        replace_config(
-            REPOSITORIES_BASE_PATH
-            / "fr.obeo.dsl.viewpoint.collab"
-            / "repository.properties",
-            repo,
-            rf"tcp\://{t4c_host}\:{t4c_port}/{repo}",
-        )
+    if t4c_json:
+        t4c_repos: list[dict[str, str]] = json.loads(t4c_json)
+        duplicate_names = [
+            name
+            for name, count in Counter(
+                [repo["repository"] for repo in t4c_repos]
+            ).items()
+            if count > 1
+        ]
+        for repo in t4c_repos:
+            replace_config_repositories(
+                f"{repo['repository']}\\ ({repo['instance']})"
+                if repo["repository"] in duplicate_names
+                else repo["repository"],
+                rf"tcp\://{repo['host']}\:{repo['port']}/{repo['repository']}",
```

Comment:
This line is the same for both parts. I'd pass `t4c_host`, `t4c_port` and `repo` to the function.